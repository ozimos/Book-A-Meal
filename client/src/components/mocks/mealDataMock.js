export const rows = [
  { id: 'abc', price: 1200 }, { id: 'def', price: 2000 }
];

export const allMeals = {
  data: { rows }

};

export const meal = {
  data: { id: 'mealId',
    price: 200,
    description: 'nice',
    imageUrl: 'nice',
    title: 'nice'
  }
};
