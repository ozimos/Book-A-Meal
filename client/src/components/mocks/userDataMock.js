export const userData = {
  data: { id: '123', email: 'asdf' },
  message: 'abc',
  token: 'abc'
};
export const userInput = {
  data: { email: 'asdf' }
};
export const userIncludes = {
  email: 'asdf',
  firstName: 'first',
  lastName: 'last'
};
