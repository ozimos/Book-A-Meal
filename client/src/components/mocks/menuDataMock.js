import { allMeals } from './mealDataMock';

export default {
  data: { Meals: allMeals.data }
};