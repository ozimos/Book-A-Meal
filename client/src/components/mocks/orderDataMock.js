import { allMeals } from './mealDataMock';

const Meals = allMeals.data.map(meal => ({ ...meal, MealOrders: { quantity: 2 } }));
export const order = {
  data: { id: 'orderId',
    userId: 'userId',
    Meals,
    mealList: ['abc', 'def'],
    quantityList: [2, 2] }
};
export const allOrders = {
  data: [{ ...order.data, id: 'order1' }, { ...order.data, id: 'order2' }]
};