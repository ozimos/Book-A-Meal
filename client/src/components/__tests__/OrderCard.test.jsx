/* global React:false, shallow:false toJson:false */

import OrderCard from '../mealCard/OrderCard';
import { meal } from '../mocks/mealDataMock';

describe('OrderCard Component', () => {
  it('should render without throwing an error', () => {
    expect(shallow(<OrderCard
      {...meal.data}
      dispatch={jest.fn()}
      addToOrder={jest.fn()}
      removeFromOrder={jest.fn()}
    />)
      .exists(<div className="card" />)).toBe(true);
  });
  it('renders correctly', () => {
    const wrapper = shallow(<OrderCard
      {...meal.data}
      dispatch={jest.fn()}
      addToOrder={jest.fn()}
      removeFromOrder={jest.fn()}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

