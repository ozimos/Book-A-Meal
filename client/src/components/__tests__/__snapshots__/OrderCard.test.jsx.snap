// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderCard Component renders correctly 1`] = `
<React.Fragment>
  <div
    className="card card-booking pillar"
  >
    <img
      alt="Meal"
      className="fluid-img"
      src="nice"
    />
    <div
      className="pillar"
    >
      <div
        className="title-element text_left long_string"
      >
        nice
      </div>
      <div
        className="flexbox"
      >
        <div
          className="text_left pad"
        >
          ₦
          200
        </div>
        <button
          className="btn title-button row_button"
          onClick={[Function]}
        >
          <div
            className="long_string2"
          >
            Add to Cart
          </div>
        </button>
        <button
          className="btn title-button icon"
          onClick={[Function]}
        >
          ✖
        </button>
      </div>
    </div>
  </div>
</React.Fragment>
`;
